### ISSUE TITLE
Software update notification

### ISSUE THREAD

**Issue by GreenLunar** [QupZilla/qupzilla#1882]
Add an auto update checker and a menu item under menu About to check for updates.

Once Qupzilla detects an update, it would prompt user to go to Qupzilla homepage, similarly to SMPlayer.

For Windows users, Qupzilla can auto-download and auto-install Qupzilla, but I am not in favor of this somewhat intrusive action.

**Comment by jamesqf** [comment]
Please no, or at least give the user an option to turn it off.  I don't want updates breaking things or making my (customized) user interfaces unusable - things which have happened with updates of other software.

**Comment by ke7ofi** [comment]
This is what package managers are for. I'd only recommend including this in the Windows builds (if possible).

**Comment by GreenLunar** [comment]
> I'd only recommend including this in the Windows builds

On any non-Linux platform that is not shipped not have a package manager (i.e. Mac OS X and Windows).

**Comment by NootanR** [comment]
Sir my qupzilla is not running fine

**Comment by ke7ofi** [comment]
Aren't there a couple of package managers for OS X (or, for that matter, Windows)? If so, maybe just have a special standalone version which updates itself.

**Comment by nowrep** [comment]
Actually I've been thinking about completely remove the update check. Because as it is now, it just shows notification that there is a new version but user must manually go to QupZilla website and download it.
Another problem is that this really works only for Windows. On Linux it depends on what distribution is user using, so the new version may not be available right after the release or not at all.

**Comment by GreenLunar** [comment]
Perhaps QupZilla (Windows) should have a separate binary file for update check, similarly to iTunes.

**Comment by nowrep** [comment]
Yes, that would be better. It could also handle the complete update process.
But I won't do it, so patches welcome :)
